thinc/backends/ops.py

```diff
@@ -569,13 +569,13 @@ def as_contig(self, data: ArrayT, dtype: Optional[DTypes] = None) -> ArrayT:
     def sigmoid(self, X: FloatsType, *, inplace: bool = False) -> FloatsType:
         if inplace:
             self.xp.exp(-X, out=X)
-            X += 1.0 # type: ignore
+            X += 1.0 # type: ignore 
             X **= -1.0 # type: ignore
             return cast(FloatsType, X)
         else:
             return cast(FloatsType, 1.0 / (1.0 + self.xp.exp(-X)))
 
-    def dsigmoid(self, Y: FloatsType, *, inplace: bool = False) -> FloatsType:
+    def dsigmoid(self, Y: FloatsT, *, inplace: bool = False) -> FloatsT:
```

Comment:
```suggestion
    def dsigmoid(self, Y: FloatsType, *, inplace: bool = False) -> FloatsType:
```
Let's revert this back so this PR focuses on the new functions. 

@honnibal for some context: we've been discussing some of the type annotations throughout the `ops` module here, and the best way forward to deal with some of the remaining `mypy` issues. I suggested to discuss this on a separate PR though, and focus on the actual implementations here.